### PR TITLE
[SPARK-36554][SQL][PYTHON] Expose make_date expression in functions.scala

### PR DESCRIPTION
**Pull request by nicolasazrak**
Hi, this is my first PR to spark, let me know if I forgot anything! 

### What changes were proposed in this pull request?

This commit solves `SPARK-36554`. It exposes the `make_date` sql expression into the scala functions file to be able to use it in expressions.

### Does this PR introduce _any_ user-facing change?

It exposes the `make_date` function to use as an expression without the need to use `expr("make_date(...)")`.

### How was this patch tested?

It has an unit test in the `test_functions.py` file.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
We will have to add it to SparkR side too but could be done separately. Also, you will have to enable Github Actions in your forked repository (see also https://github.com/apache/spark/pull/34356/checks?check_run_id=3964321653).

**Comment by HyukjinKwon** [comment]
ok to test

**Comment by nicolasazrak** [comment]
@yoda-mon @HyukjinKwon @sarutak  can we merge this?

File: python/pyspark/sql/functions.py [hunk 1]
```diff
@@ -2131,6 +2131,32 @@ def weekofyear(col: "ColumnOrName") -> Column:
...
 
+def make_date(year: "ColumnOrName", month: "ColumnOrName", day: "ColumnOrName") -> Column:
+    """
+    Returns a column with a date built from the year, month and day columns.
+
+    .. versionadded:: 3.3.0
+
+    Parameters
+    ----------
+    year : :class:`ColumnOrName` with the year to build the date
```
Comment (sarutak) [inline python/pyspark/sql/functions.py:2142]: Type hints should be `ColumnOrName` but the type for each parameter should be ``` `~pyspark.sql.Column` or str ``` here.
Also, please place the description of each parameter in the next line with indentation like
```
year : class:`~pyspark.sql.Column` or str
    The year to build the date.
```

**Comment by sarutak** [comment]
@nicolasazrak Please change `pyspark.sql.rst` together whenever you add APIs for PySpark.
Also, could you make sure that the API docs are successfully build and the layout is not broken.
See also http://spark.apache.org/docs/latest/api/python/development/contributing.html#contributing-documentation-changes

**Comment by sarutak** [comment]
Confirmed the API doc is built successfully with `make html`.
make_date_doc

**Review by sarutak — APPROVED** [review]

**Comment by sarutak** [comment]
Merged to `master`. Thanks, all !

**Comment by HyukjinKwon** [comment]
👍

**Pull request closed without merging**